### PR TITLE
Issue 94

### LINKED ISSUE #94: Check quakeml file lengths
Related to this java bug report:
https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8222187

Files that are exactly (X*4096 + 1) bytes long, trigger this bug when using a buffer size of 4096...  Check for a file that is this long and pad with one extra space before processing.

### PR DESCRIPTION
**Pull request by Capprin**
Code to fix null byte bug: https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8222187

closes #94

### CODE REVIEW

**Comment by jmfee-usgs** [comment]
Please fix this in a different way, rather than updating the file before processing...

Add a utility method to the QuakemlProductCreator that does this check and pads if needed. 
 Also add a utility method that does the formatConverter.getString call, and update all places that use the getString method to use that method (and the other location that should be using formatConverter but creates a new Converter() object).

That will cover most cases, but you should also check for the case where rawQuakeml is not null and use the first utility method to apply padding if needed:
https://github.com/usgs/pdl/blob/cff967462df1767fedcfabc920754333373d8c2c/src/main/java/gov/usgs/earthquake/eids/QuakemlProductCreator.java#L124


Also, rebase.

File: src/main/java/gov/usgs/earthquake/eids/EIDSInputWedge.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package gov.usgs.earthquake.eids;
 
+import java.io.BufferedWriter;
```
Comment (jmfee-usgs) [inline src/main/java/gov/usgs/earthquake/eids/EIDSInputWedge.java]: These imports are no longer needed.

File: src/main/java/gov/usgs/earthquake/eids/QuakemlProductCreator.java [hunk 2]
```diff
@@ -89,6 +89,16 @@
...
 	private boolean validate = false;
 
+	private boolean padForBase64Bug = false;
+
+	public QuakemlProductCreator() {
+		super();
+	}
+
+	public QuakemlProductCreator(boolean padForBase64Bug) {
+		this.padForBase64Bug = padForBase64Bug;
```
Comment (jmfee-usgs) [inline src/main/java/gov/usgs/earthquake/eids/QuakemlProductCreator.java]: Another option is to  add getter/setter methods to change the default behavior.

File: src/main/java/gov/usgs/earthquake/eids/QuakemlProductCreator.java [hunk 3]
```diff
@@ -880,6 +890,33 @@ public boolean isSendMechanismWhenPhasesExist() {
...
+	 * Utility function that converts quakeml to a string
+	 *
+	 * @param message
+	 * 						The quakeml to be converted
+	 *
+	 * @return raw string
+	 * @throws Exception if quakeml doesn't validate
+	 */
+	private String convertQuakemlToString(Quakeml message) throws Exception{
+		return formatConverter.getString(message,validate);
```
[the remark this answers is not in the document]
Comment (Capprin) [reply src/main/java/gov/usgs/earthquake/eids/QuakemlProductCreator.java]: oops.

File: src/main/java/gov/usgs/earthquake/eids/QuakemlProductCreator.java (hunk 2 above)
Comment (Capprin) [reply src/main/java/gov/usgs/earthquake/eids/QuakemlProductCreator.java]: Left the constructor, but added getters and setters so users can implement however they want

**Comment by Capprin** [comment]
The documentation still hasn't been updated, but I'm not sure of a good place to put this new parameter. Please advise.

**Pull request merged**